aptos-move/framework/src/natives/event.rs

```diff
@@ -50,11 +90,20 @@ fn native_write_to_event_store(
         EVENT_WRITE_TO_EVENT_STORE_BASE
             + EVENT_WRITE_TO_EVENT_STORE_PER_ABSTRACT_VALUE_UNIT * context.abs_val_size(&msg),
     )?;
+    let ty_tag = context.type_to_type_tag(&ty)?;
+    let ty_layout = context.type_to_type_layout(&ty)?;
+    let blob = msg.simple_serialize(&ty_layout).ok_or_else(|| {
+        SafeNativeError::InvariantViolation(PartialVMError::new(
+            StatusCode::VALUE_DESERIALIZATION_ERROR,
```

Comment:
Please change to another status code under the invariant violation category. `SafeNativeError::InvariantViolation` is only supposed to take invariant violations. Use `UNKNOWN_INVARIANT_VIOLATION_ERROR` if you are not sure what to use.